clients/spark/src/main/scala/io/treeverse/clients/S3ClientBuilder.scala

```diff
@@ -93,17 +93,14 @@ object S3ClientBuilder extends S3ClientBuilder {
       }
 
     val credentialsProvider: AWSCredentialsProvider =
-      if (roleArn.nonEmpty && !useHadoopProvider) {
+      if (roleArn.nonEmpty && (!wantHadoopAssume || !hadoopAssumeAvailable)) {
```

Comment:
we want to assume role if the environment is configured to assume and we can't use the hadoop implementation and the role arn is configured.